subsys/bluetooth/host/buf.c

```diff
@@ -147,3 +150,66 @@ struct net_buf_pool *bt_buf_get_num_complete_pool(void)
 }
 #endif /* CONFIG_BT_CONN || CONFIG_BT_ISO */
 #endif /* ZTEST_UNITTEST */
+
+struct net_buf *bt_buf_make_view(struct net_buf *view,
+				 struct net_buf *parent,
+				 size_t len,
+				 struct bt_buf_view_meta *meta)
+{
+	__ASSERT_NO_MSG(len);
+	__ASSERT_NO_MSG(view);
+	/* The whole point of this API is to allow prepending data. If the
+	 * headroom is 0, that will not happen.
+	 */
+	__ASSERT_NO_MSG(net_buf_headroom(parent) > 0);
+
+	/* `parent` should have been just re-used instead of trying to make a
+	 * view into it.
+	 */
+	__ASSERT_NO_MSG(len < parent->len);
+
+	__ASSERT_NO_MSG(!bt_buf_has_view(parent));
+
+	LOG_DBG("make-view %p viewsize %u meta %p", view, len, meta);
+
+	/* The ref to `parent` is moved in by passing `parent` as argument. */
```

Comment:
This comment is misplaced? It should just go in the doxygen, I think.

Reply:
Or it should go with line https://github.com/zephyrproject-rtos/zephyr/pull/72090/files#diff-832d014070781267cbe71bb402563daa5799698fb1212b2883f90a1fc485fbeaR185